Add docker install doc (#1227)
grafana/loki: docs/installation/docker.md

# Installing Loki with Docker/Docker-compose

## Prerequisites

> Basiclly, you must install docker.

Reviewer: Could you link to Docker's installation website here?

> Basiclly, you must install docker.

Reviewer: ```suggestion
Basiclly, you must install Docker.
```

> If you want to install loki with docker-compose, install it too.

Reviewer: Same thing here: could you link to where to install docker-compose?

> If you want to install loki with docker-compose, install it too.

Reviewer: ```suggestion
If you want to install Loki with docker-compose, install it too.
```

## Install with Docker

```bash
$ wget https://raw.githubusercontent.com/grafana/loki/master/cmd/loki/loki-local-config.yaml -o loki-config.yaml
$ docker run -p 3100:3100 grafana/loki:latest -config.file=./loki-config.yaml
$ wget https://raw.githubusercontent.com/grafana/loki/master/cmd/promtail/promtail-docker-config.yaml -o promtail-config.yaml

> $ docker run -v /var/log:/var/log grafana/promtail:latest -config.file=./promtail-config.yaml

Reviewer: Few things here:

1. Even though the files are being downloaded, they're not being sent to the container. A volume mount is needed. something like `-v $(pwd):/mnt/config -config.file=/mnt/config/loki-config.yaml` 
2. Rather than using latest, I think we should refer to the latest release tag here. We'll keep it up to date as part of our release process.

```

## Install with Docker-compose

```bash
$ wget https://raw.githubusercontent.com/grafana/loki/master/production/docker-compose.yaml -o docker-compose.yaml

> # docker-compose -f docker-compose.yaml up

Reviewer: Does the `#` here imply sudo? I think it should just be `$` like the other line; most people have their docker configured to not need sudo access, and those who do should know to insert it.

> # Installing Loki with Docker/Docker-compose

Reviewer: ```suggestion
# Installing Loki with Docker or Docker Compose
```

> You can installing Loki with Docker/Docker-compose for try, test or development.

Reviewer: ```suggestion
You can install Loki with Docker or Docker Compose for evaluating, testing, or developing Loki.
```

For production, we recommend Tanka or Helm.

## Prerequisites

Basically, you must install [docker](https://docs.docker.com/install).

> If you want to install Loki with [docker-compose](https://docs.docker.com/compose/install/), install it too.

Reviewer: We can just make this a list: 

```markdown
- [Docker](https://docs.docker.com/install)
- [`docker-compose`](https://docs.docker.com/compose/install) (optional, only needed for the Docker Compose install method) 
```

## Install with Docker

```bash
$ wget https://raw.githubusercontent.com/grafana/loki/v1.2.0/cmd/loki/loki-local-config.yaml -o loki-config.yaml
$ docker run -v $(pwd):/mnt/config -config.file=/mnt/config/loki-config.yaml -p 3100:3100 grafana/loki:v1.2.0
$ wget https://raw.githubusercontent.com/grafana/loki/v1.2.0/cmd/promtail/promtail-docker-config.yaml -o promtail-config.yaml

> $ docker run -v $(pwd):/mnt/config -v /var/log:/var/log -config.file=/mnt/config/promtail-config.yaml grafana/promtail:latest

Reviewer: I'm not sure this is right either. Don't the arguments to the container have to go after the container name?

```

> ## Install with Docker-compose

Reviewer: ```suggestion
## Install with Docker Compose
```
